src/auth/awsclient.ts

```diff
@@ -335,4 +337,21 @@ export class AwsClient extends BaseExternalAccountClient {
     });
     return response.data;
   }
+
+  private shouldUseMetadataServer(): boolean {
+    return (
+      !this.canRetrieveRegionFromEnvironment() ||
+      !this.canRetrieveSecurityCredentialsFromEnvironment()
+    );
+  }
+
+  private canRetrieveRegionFromEnvironment(): boolean {
+    return !!(process.env['AWS_REGION'] || process.env['AWS_DEFAULT_REGION']);
+  }
+
+  private canRetrieveSecurityCredentialsFromEnvironment(): boolean {
+    return !!(
+      process.env['AWS_ACCESS_KEY_ID'] && process.env['AWS_SECRET_ACCESS_KEY']
+    );
```

Comment:
Same here, would be more useful as a [getter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get).